src/polydb.jl

```diff
@@ -0,0 +1,308 @@
+# returns two arrays containing the names of the meta data collections
+# first one for sections, second one for collections
+function _get_meta_names(names::Array{String, 1})
+   n = length(names)
+   sec_bool = BitArray{1}(undef, n)
+   coll_bool = BitArray{1}(undef, n)
+   n_secs = 0
+   n_colls = 0
+   i = 1
+   for name in names
+      if SubString(name, 1, 2) == "_s"
+         sec_bool[i] = true
+         coll_bool[i] = false
+         n_secs += 1
+      elseif SubString(name, 1, 2) == "_c"
+         sec_bool[i] = false
+         coll_bool[i] = true
+         n_colls += 1
+      else
+         sec_bool[i] = false
+         coll_bool[i] = false
+      end
+      i += 1
+   end
+   secs = Base.Array{String,1}(undef, n_secs)
+   colls = Base.Array{String,1}(undef, n_colls)
+   i_s = 1
+   i_c = 1
+   for j = 1:n
+      if sec_bool[j]
+         secs[i_s] = names[j]
+         i_s += 1
+      elseif coll_bool[j]
+         colls[i_c] = names[j]
+         i_c += 1
+      end
+   end
+   return secs, colls
+end
+
+# functions helping printing metadata for sections or collections
+function _get_contact(s::String)
+   return s
+end
+
+function _get_contact(a::Array)
+   res = ""
+   for dict in a
+      str = dict["name"]
+      for key in ["email", "www", "affiliation"]
+         if !isempty(get(dict, key, ""))
+            str = string(str, ", ", dict[key])
+         end
+      end
+      res = string(res, "\t\t", str, "\n")
+   end
+   return res
+end
+
+# prints information about a specific section and
+# continues to print information about its content
+function _get_section(db::Database, info::Mongoc.BSON, sections::Array{String,1}, collections::Array{String,1})
+   res = string("SECTION: ", join(info["section"], "."), "\n", info["description"], "\n")
+   if haskey(info, "maintainer")
+      res = string(res, "Maintained by ", info["maintainer"]["name"], ", ", info["maintainer"]["email"], ", ", info["maintainer"]["affiliation"], "\n")
+   end
+   return string(res, "\n", _get_sections(db, info["section"], sections, collections))
+end
+
+# prints information about a specific collection
+function _get_collection(info::Mongoc.BSON)
+   res = string("\tCOLLECTION: ", join(info["section"], "."), ".", info["collection"], "\n")
+   if haskey(info, "description")
+      res = string(res, "\t", info["description"], "\n")
+   end
+   if haskey(info, "author")
+      res = string(res, "\tAuthored by ", "\n", _get_contact(info["author"]))
+   end
+   if haskey(info, "maintainer")
+      res = string(res, "\tMaintained by", "\n", _get_contact(info["maintainer"]))
+   end
+   return res
+end
+
+# initializes printing complete section/collection tree
+function _get_sections(db::Database, sections::Array{String,1}, collections::Array{String,1})
+   res = ""
+   for sec in sections
+      sec_c = db.mdb[sec]
+      info = iterate(sec_c)[1]
+      if length(info["section"]) == 1
+         res = string(res, _get_section(db, info, sections, collections))
```

Comment:
probably this is not crucial here, but every time we allocate another string that grows in length.
It is much better to construct all of them (in an array) and then `join` them at the end